src/main/java/com/icthh/xm/ms/entity/domain/listener/XmEntityElasticSearchListener.java

```diff
@@ -23,6 +28,12 @@
     private static XmEntitySearchRepository xmEntitySearchRepository;
 
     private static XmEntitySpecService xmEntitySpecService;
+    private static XmEntityElasticRepository xmEntityElasticRepository;
+
+    @Autowired
+    public void setXmEntityElasticRepository(XmEntityElasticRepository xmEntityElasticRepository) {
+        XmEntityElasticSearchListener.xmEntityElasticRepository = xmEntityElasticRepository;
```

Comment:
Elastic it's implementation details, please use search repository at this level

Reply:
I am not sure what do u mean by "implementation details".  Let me explaine  what i did here a bit more
Here i have 2 different ways of handling reindex. 
 - first is a `XmEntityMultipleIndexElasticIndexRepository `- the new one. It is based on `ElasticsearchTemplate `and not on `XmEntitySearchRepository`, coz it actually knows how to create index and has bulk operations. 
 - and seconds one `XmEntitySingleIndexElasticRepository`, which repeats existing functionality and based on `XmEntitySearchRepository`.
I this i should have removed `XmEntitySearchRepository` from `XmEntityElasticSearchListener` and `ProfileElasticSearchListener`, coz it is not used there anymore and this is what makes some confusion.

Reply:
@amedvedchuk  pls give your point of view

Reply:
hi @ibilous 
by 'implementation details' mentioned by @sergeysenja1992  he means the name of the class `XmEntityElasticRepository` which exposes implementation detail in it's name (Elastlic). 
Regarding your explanation, I suggest marking all the classes that you are planning t delete with @Deprecated and appropriate reasons. So anyone can see it from the MR.

Reply:
hi :) actually i do no plan to delete any class.  Here i just extended existed logic.
I do not think that we cannot use "Elastic" in the name of class. For example in communication service we use the word "Telegram" in class name.  I am not insist on that name but think that it is correct